engine/src/test/java/org/hibernate/validator/test/cfg/ValidatorConstraintMappingTest.java

```diff
@@ -0,0 +1,308 @@
+	}
+
+	@Test(
+			expectedExceptions = IllegalArgumentException.class,
+			expectedExceptionsMessageRegExp = "HV[0-9]*: The annotation type must not be null when creating a constraint definition."
+	)
+	public void testNullClass() {
+		mapping.constraint( null );
+	}
+
+	@Test(
+			expectedExceptions = IllegalArgumentException.class,
+			expectedExceptionsMessageRegExp = "HV[0-9]*: The annotation type must be annotated with @javax.validation.Constraint when creating a constraint definition."
+	)
+	public void testNonConstraintAnnotation() {
+		mapping.constraint( NonConstraintAnnotation.class );
+	}
+
+	@Test
+	public void testConstraintMapping() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.validatedBy( NonDefaultLongValidator.class );
+
+		config.addMapping( mapping );
+		Validator validator = config.buildValidatorFactory().getValidator();
+
+		Set<? extends ConstraintViolation<?>> violations = validator.validate( new ConstrainedLongFieldBean() );
+		assertNumberOfViolations( violations, 1 );
+		assertCorrectValidatorTypes( violations, NonDefaultLongValidator.class );
+	}
+
+	@Test
+	public void testConstraintMappingDefaultsToIncludingExistingValidators() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.validatedBy( NonDefaultLongValidator.class );
+
+		config.addMapping( mapping );
+		Validator validator = config.buildValidatorFactory().getValidator();
+
+		Set<? extends ConstraintViolation<?>> violations = validator.validate( new ConstrainedStringFieldBean() );
+		assertNumberOfViolations( violations, 1 );
+		assertCorrectValidatorTypes( violations, DefaultStringValidator.class );
+	}
+
+	@Test
+	public void testConstraintMappingIncludingExistingValidators() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.includeExistingValidators( true )
+				.validatedBy( NonDefaultLongValidator.class );
+
+		config.addMapping( mapping );
+		Validator validator = config.buildValidatorFactory().getValidator();
+
+		Set<? extends ConstraintViolation<?>> violations = validator.validate( new ConstrainedStringFieldBean() );
+		assertNumberOfViolations( violations, 1 );
+		assertCorrectValidatorTypes( violations, DefaultStringValidator.class );
+	}
+
+	@Test
+	public void testConstraintMappingExcludingExistingValidators() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.includeExistingValidators( false )
+				.validatedBy( NonDefaultIntegerValidator.class );
+
+		config.addMapping( mapping );
+		Validator validator = config.buildValidatorFactory().getValidator();
+
+		Set<? extends ConstraintViolation<?>> violations = validator.validate( new ConstrainedIntegerFieldBean() );
+		assertNumberOfViolations( violations, 1 );
+		assertCorrectValidatorTypes( violations, NonDefaultIntegerValidator.class );
+	}
+
+	@Test
+	public void testConstraintMappingIncludingExistingValidatorsThenExcludingThem() {
+		mapping.constraint( ConstraintAnnotation.class )
+				.includeExistingValidators( true )
```

Comment:
Agreed, only one call to `includeExistingValidators()` would make sense. But still, it's possible to call it multiple times, so the behavior should be defined and tested.
I am not considering the (rather complex) behavior you just described.
In this test, I'm simply ensuring that calling `.includeExistingValidators( true ).includeExistingValidators( false )` produces the exact same effect than just calling `.includeExistingValidators( false )`. So I just test that 
What I tried to say in my previous message is that, if there was a bug in the tested code, and somehow the first value (true) was taken into account instead of the second (false), then the test would throw an unexpected exception because we would have two validators for the same type. So the test would fail.